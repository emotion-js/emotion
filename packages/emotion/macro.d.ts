declare module 'emotion/macro' {
  export * from 'emotion'
}
