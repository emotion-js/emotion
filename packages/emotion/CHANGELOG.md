# emotion

## 10.0.27

### Patch Changes

- [`4c62ae9`](https://github.com/emotion-js/emotion/commit/4c62ae9447959d438928e1a26f76f1487983c968) [#1698](https://github.com/emotion-js/emotion/pull/1698) Thanks [@Andarist](https://github.com/Andarist)! - Add LICENSE file
- Updated dependencies [[`4c62ae9`](https://github.com/emotion-js/emotion/commit/4c62ae9447959d438928e1a26f76f1487983c968)]:
  - babel-plugin-emotion@10.0.27
  - create-emotion@10.0.27
  - @emotion/stylis@0.8.5

## 10.0.23

### Patch Changes

- [`1ae94891`](https://github.com/emotion-js/emotion/commit/1ae948917326e2bc2bc978c59d04cb2695c72e38) [#1583](https://github.com/emotion-js/emotion/pull/1583) Thanks [@Andarist](https://github.com/Andarist)! - Export TS & Flow types for emotion/macro

- Updated dependencies [[`3927293d`](https://github.com/emotion-js/emotion/commit/3927293d0b9d96b4a7c00196e8430728759b1161), [`b3a0f148`](https://github.com/emotion-js/emotion/commit/b3a0f1484f2efcc78b447639ff2e0bc0f29915ae)]:
  - babel-plugin-emotion@10.0.23

## 10.0.17

### Patch Changes

- [66cda641](https://github.com/emotion-js/emotion/commit/66cda64128631790b81e3c9df273a972358ea593) [#1478](https://github.com/emotion-js/emotion/pull/1478) Thanks [@Andarist](https://github.com/Andarist)! - Add warnings about using illegal escape sequences

## 10.0.14

### Patch Changes

- [c0eb604d](https://github.com/emotion-js/emotion/commit/c0eb604d) [#1419](https://github.com/emotion-js/emotion/pull/1419) Thanks [@mitchellhamilton](https://github.com/mitchellhamilton)! - Update build tool
