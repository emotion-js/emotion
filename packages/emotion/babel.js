module.exports = require('babel-plugin-emotion')
