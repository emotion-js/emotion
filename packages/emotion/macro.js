module.exports = require('babel-plugin-emotion').macros.emotion
