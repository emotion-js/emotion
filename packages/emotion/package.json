{
  "name": "emotion",
  "version": "9.2.4",
  "description": "The Next Generation of CSS-in-JS.",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "types": "types/index.d.ts",
  "files": [
    "src",
    "dist",
    "macro.js",
    "types"
  ],
  "scripts": {
    "build": "npm-run-all clean rollup",
    "test:typescript": "dtslint types",
    "clean": "rimraf dist",
    "rollup": "rollup -c ../../rollup.config.js",
    "watch": "rollup -c ../../rollup.config.js --watch"
  },
  "dependencies": {
    "babel-plugin-emotion": "^9.2.4",
    "create-emotion": "^9.2.4"
  },
  "devDependencies": {
    "@types/react": "16.0.16",
    "babel-cli": "^6.24.1",
    "babel-plugin-transform-define": "^1.3.0",
    "cross-env": "^5.0.5",
    "dtslint": "^0.3.0",
    "npm-run-all": "^4.0.2",
    "rimraf": "^2.6.1",
    "rollup": "^0.60.1"
  },
  "author": "Kye Hohenberger",
  "homepage": "https://emotion.sh",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/emotion/tree/master/packages/emotion",
  "keywords": [
    "styles",
    "emotion",
    "react",
    "css",
    "css-in-js"
  ],
  "bugs": {
    "url": "https://github.com/emotion-js/emotion/issues"
  }
}
