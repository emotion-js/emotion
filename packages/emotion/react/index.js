module.exports = require('react-emotion')
