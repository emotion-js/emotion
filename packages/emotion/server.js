module.exports = require('emotion-server')
