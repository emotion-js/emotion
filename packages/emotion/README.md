<p align="center" style="color: #343a40">
  <img src="https://cdn.rawgit.com/tkh44/emotion/master/emotion.png" alt="emotion" height="150" width="150">
  <h1 align="center">emotion</h1>
</p>
<p align="center" style="font-size: 1.2rem;">The Next Generation of CSS-in-JS</p>

[![Backers on Open Collective](https://opencollective.com/emotion/backers/badge.svg)](#backers) [![Sponsors on Open Collective](https://opencollective.com/emotion/sponsors/badge.svg)](#sponsors) [![npm version](https://badge.fury.io/js/emotion.svg)](https://badge.fury.io/js/emotion)
[![Build Status](https://img.shields.io/circleci/project/github/emotion-js/emotion/master.svg)](https://circleci.com/gh/emotion-js/emotion)
[![codecov](https://codecov.io/gh/emotion-js/emotion/branch/master/graph/badge.svg)](https://codecov.io/gh/emotion-js/emotion)
![core gzip size](http://img.badgesize.io/https://unpkg.com/emotion/dist/emotion.umd.min.js?compression=gzip&label=core%20gzip%20size)
![core size](http://img.badgesize.io/https://unpkg.com/emotion/dist/emotion.umd.min.js?label=core%20size)
![react gzip size](http://img.badgesize.io/https://unpkg.com/react-emotion/dist/emotion.umd.min.js?compression=gzip&label=react%20gzip%20size)
![react size](http://img.badgesize.io/https://unpkg.com/react-emotion/dist/emotion.umd.min.js?label=react%20size)
[![slack](https://emotion.now.sh/badge.svg)](http://emotion.now.sh/)

Emotion is a performant and flexible CSS-in-JS library. Building on many other CSS-in-JS libraries, it allows you to style apps quickly with string or object styles. It has predictable composition to avoid specificity issues with CSS. With source maps and labels, Emotion has a great developer experience and great performance with heavy caching in production.

# [üëÄ Demo Sandbox](https://codesandbox.io/s/pk1qjqpw67)

# [üìñ Docs](https://emotion.sh/docs/introduction)

Frequently viewed docs:

* [Introduction](https://emotion.sh/docs/introduction)
* [Install](https://emotion.sh/docs/install)
* [CSS](https://emotion.sh/docs/css)
* [Styled Components](https://emotion.sh/docs/styled)
* [Composition](https://emotion.sh/docs/composition)
* [Nested Selectors](https://emotion.sh/docs/nested)
* [Media Queries](https://emotion.sh/docs/media-queries)

### Quick Start

Get up and running with a single import.

```bash
npm install --save emotion
```

```javascript
import { css } from 'emotion'

const app = document.getElementById('root')
const myStyle = css`
  color: rebeccapurple;
`
app.classList.add(myStyle)
```

### React with [Optional Babel Plugin](docs/babel.md)

```bash
npm install --save emotion react-emotion babel-plugin-emotion
```

_Note: use `preact-emotion` in place of `react-emotion` if using [Preact](https://github.com/developit/preact)_

```javascript
import styled, { css } from 'react-emotion'

const Container = styled('div')`
  background: #333;
`
const myStyle = css`
  color: rebeccapurple;
`
const app = () => (
  <Container>
    <p className={myStyle}>Hello World</p>
  </Container>
)
```

### Do I Need To Use the Babel Plugin?

The babel plugin is not required, but enables some optimizations and customizations that could be beneficial for your project.

Look here üëâ _[emotion babel plugin feature table and documentation](https://github.com/emotion-js/emotion/tree/master/packages/babel-plugin-emotion)_

### Demo Sandbox

[Demo Code Sandbox](https://codesandbox.io/s/pk1qjqpw67)

### Examples

* [emotion website](site) [[Demo Here](https://emotion.sh)]
* [next-hnpwa-guide-kit](https://github.com/tkh44/next-hnpwa-guide-kit) [[Demo Here](https://hnpwa.life)]
* [reactivesearch](https://github.com/appbaseio/reactivesearch), a react UI library for Elasticsearch [[Website](https://opensource.appbase.io/reactivesearch/)]
* [circuit-ui](https://github.com/sumup/circuit-ui), a react component library built at SumUp [[Storybook](https://sumup.github.io/circuit-ui/)]
* [govuk-react](https://github.com/penx/govuk-react/), a React component library built for UK Government departments
* **open a PR and add yours!**

### Ecosystem

* [facepaint](https://github.com/emotion-js/facepaint)
* [emotion-vue](https://github.com/egoist/emotion-vue)
* [ember-emotion](https://github.com/alexlafroscia/ember-emotion)
* [CSS to emotion transform](https://transform.now.sh/css-to-emotion/)
* [ShevyJS](https://github.com/kyleshevlin/shevyjs)
* [design-system-utils](https://github.com/mrmartineau/design-system-utils) - Utilities to give better access to your design system.

### In the Wild

* [healthline.com](https://www.healthline.com)
* [nytimes.com](https://www.nytimes.com)
* [vault.crucible.gg](http://vault.crucible.gg/)
* [saldotuc.com](https://saldotuc.com)
* [gatsbythemes.com](https://gatsbythemes.com/)
* [blazity.com](https://blazity.com/)
* [postmates.com](https://postmates.com/)
* [thedisconnect.co](https://thedisconnect.co/one)
* [zefenify.com](https://zefenify.com/about.html)
* [sentry.io](https://sentry.io)

## Contributors

This project exists thanks to all the people who contribute. [[Contribute](CONTRIBUTING.md)].
<a href="graphs/contributors"><img src="https://opencollective.com/emotion/contributors.svg?width=890&button=false" /></a>

## Backers

Thank you to all our backers! üôè [[Become a backer](https://opencollective.com/emotion#backer)]

<a href="https://opencollective.com/emotion#backers" target="_blank"><img src="https://opencollective.com/emotion/backers.svg?width=890"></a>

## Sponsors

Support this project by becoming a sponsor. Your logo will show up here with a link to your website. [[Become a sponsor](https://opencollective.com/emotion#sponsor)]

<a href="https://opencollective.com/emotion/sponsor/0/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/0/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/1/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/1/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/2/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/2/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/3/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/3/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/4/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/4/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/5/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/5/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/6/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/6/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/7/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/7/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/8/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/8/avatar.svg"></a>
<a href="https://opencollective.com/emotion/sponsor/9/website" target="_blank"><img src="https://opencollective.com/emotion/sponsor/9/avatar.svg"></a>
