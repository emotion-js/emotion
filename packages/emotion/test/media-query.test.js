import React from 'react'
import renderer from 'react-test-renderer'
import { css, sheet } from 'emotion'

describe('css', () => {
  test('handles media query merges', () => {
    const mq = [
      '@media(min-width: 420px)',
      '@media(min-width: 640px)',
      '@media(min-width: 960px)'
    ]
    const buttonCSS = [
      {
        color: 'red',
        [mq[1]]: {
          color: 'blue'
        }
      },
      {
        color: 'purple',
        [mq[1]]: {
          color: 'aquamarine'
        }
      }
    ]

    const cls1 = css([
      {
        color: 'darkslateblue',
        [mq[0]]: {
          color: 'amethyst'
        },
        [mq[1]]: {
          color: 'rebeccapurple'
        },
        [mq[2]]: {
          color: 'burlywood'
        }
      },
      buttonCSS
    ])
    const tree = renderer.create(<div className={cls1}/>).toJSON()
    expect(tree).toMatchSnapshot()
    expect(sheet).toMatchSnapshot()
  })
})
