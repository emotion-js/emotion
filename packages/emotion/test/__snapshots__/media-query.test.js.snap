// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css handles media query merges 1`] = `
.glamor-0 {
  color: darkslateblue;
  color: red;
  color: purple;
}

@media (min-width:420px) {
  .glamor-0 {
    color: amethyst;
  }
}

@media (min-width:640px) {
  .glamor-0 {
    color: rebeccapurple;
  }
}

@media (min-width:960px) {
  .glamor-0 {
    color: burlywood;
  }
}

@media (min-width:640px) {
  .glamor-0 {
    color: blue;
  }
}

@media (min-width:640px) {
  .glamor-0 {
    color: aquamarine;
  }
}

<div
  className="glamor-0"
/>
`;

exports[`css handles media query merges 2`] = `
".css-yno01n {
  color: darkslateblue;
  color: red;
  color: purple;
}

@media (min-width:420px) {
  .css-yno01n {
    color: amethyst;
  }
}

@media (min-width:640px) {
  .css-yno01n {
    color: rebeccapurple;
  }
}

@media (min-width:960px) {
  .css-yno01n {
    color: burlywood;
  }
}

@media (min-width:640px) {
  .css-yno01n {
    color: blue;
  }
}

@media (min-width:640px) {
  .css-yno01n {
    color: aquamarine;
  }
}"
`;
